optimade/server/middleware.py

```diff
@@ -65,3 +69,128 @@ async def dispatch(self, request: Request, call_next):
             self.check_url(parsed_url)
         response = await call_next(request)
         return response
+
+
+class AddWarnings(BaseHTTPMiddleware):
+    """Add any raised `OptimadeWarning`s and subclasses thereof to the response meta.warnings"""
+
+    def showwarning(
+        self,
+        message: Warning,
+        category: Type[Warning],
+        filename: str,
+        lineno: int,
+        file: Optional[IO] = None,
+        line: Optional[str] = None,
+    ) -> None:
+        """Hook to write a warning to a file."""
```

Comment:
Would you mind expanding on this docstring a bit, explaining that it comes from the stdlib, and that its main use is writing to stderr? I hadn't come across [showwarning](https://docs.python.org/3/library/warnings.html#warnings.showwarning) before